### DISCUSSION TITLE
Importer plugins for different blogs

### DISCUSSION THREAD

**Discussion by Ryuno-Ki** [11ty/eleventy#4039] [Enhancement Queue]
Inspired by [jekyll-import](https://github.com/jekyll/jekyll-import/tree/master/lib/jekyll-import/importers) I was wondering, whether it would be possible one day, to run a command and import the contents of an existing CMS.
Could ease the transition A LOT.

Next to the issues there, other JAMstacks spring to my mind: Hugo, Gatsby, Jekyll, Cactus, Hexo, Harp.js.
Haven't checked for prior art here.

**Comment by zachleat** [comment]
This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by Ryuno-Ki** [comment]
Here's a blog post about how the port from Jekyll happened: https://www.webstoemp.com/blog/from-jekyll-to-eleventy/
(leaving it here as I might want to consider it for a PR)

**Comment by Ryuno-Ki** [comment]
Concerning #465: What do you think about an approach like this: 

1. Define a config which describes the mapping between the source frontmatter and eleventy's
2. Define a config which describes the mapping between the directory structure of the source and how it should look like in eleventy.
3. Define a config file which describes the mapping between the source config file and `.eleventy.js`.

Given those information a script should be able to convert one SSG into Eleventy.

I could test it with Hexo, Hugo and Jekyll.
Am I missing something?

**Comment by zachleat** [comment]
WordPress import plugin shipped here https://www.11ty.dev/docs/migrate/
